CMakeLists.txt

```diff
@@ -27,90 +27,159 @@
+endif()
+
+# Tests currently only run on OS X @ OSRF jenkins
+# Enable on Linux by providing a display, enable on Windows via EnableDisplayTests=True
+option(EnableDisplayTests "EnableDisplayTests")
+set(DisplayTests "False" CACHE STRING "DisplayTestsVariable")
+
+if(DEFINED ENV{DISPLAY})
+  set(DISPLAYPRESENT TRUE)
+endif()
+
+if(APPLE OR DISPLAYPRESENT OR EnableDisplayTests STREQUAL "True")
+  message(STATUS "Enabling tests requiring a display")
 else()
-  message(STATUS "Using Qt5 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
-  find_package(Qt5 ${rviz_QT_VERSION} EXACT REQUIRED Core Widgets)
-  ## make target_link_libraries(${QT_LIBRARIES}) pull in all required dependencies
-  set(QT_LIBRARIES Qt5::Widgets)
-  qt5_wrap_cpp(MOC_FILES
-  src/detected_persons_display.h
-  src/tracked_persons_display.h
-  src/tracked_groups_display.h
-  src/social_relations_display.h
-  src/social_activities_display.h
-  src/human_attributes_display.h
-  src/person_display_common.h
-  src/additional_topic_subscriber.h
-)
+  set(SKIP_DISPLAY_TESTS "SKIP_TEST")
 endif()
 
-add_definitions(-DQT_NO_KEYWORDS)
+# options and directories for visual tests (see visual_testing_framework documentation)
+option(EnableVisualTests "decides whether or not to enable the tests")
 
+add_definitions(-D_BUILD_DIR_PATH="${CMAKE_CURRENT_BINARY_DIR}")
+add_definitions(-D_SRC_DIR_PATH="${CMAKE_CURRENT_SOURCE_DIR}")
 
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test_images)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test/reference_images)
+
+# Run visual tests only if "EnableVisualTests=True"
+if(EnableVisualTests STREQUAL "True")
+  message(STATUS "Enabling visual tests")
+else()
+  set(SKIP_VISUAL_TESTS "SKIP_TEST")
+endif()
+
+if(MSVC)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
+endif()
 
-set(SOURCE_FILES
+# We specifically don't turn on CMAKE_AUTOMOC, since it generates one huge
+# mocs_compilation.cpp file that takes a lot of memory to compile.  Instead
+# we create individual moc files that can be compiled separately.
+
+find_package(ament_cmake REQUIRED)
+find_package(rviz_common REQUIRED)
+find_package(rviz_rendering REQUIRED)
+find_package(rviz_default_plugins REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rviz_ogre_vendor REQUIRED)
+find_package(spencer_tracking_msgs REQUIRED)
+find_package(spencer_human_attribute_msgs REQUIRED)
+find_package(spencer_social_relation_msgs REQUIRED)
+find_package(Qt5 REQUIRED COMPONENTS Core Widgets)
+
+set(spencer_tracking_rviz_plugin_headers_to_moc 
+  include/spencer_tracking_rviz_plugin/detected_persons_display.hpp
+  include/spencer_tracking_rviz_plugin/tracked_persons_display.hpp
+  ##include/spencer_tracking_rviz_plugin/tracked_groups_display.hpp
+  ##include/spencer_tracking_rviz_plugin/social_relations_display.hpp
+  ##include/spencer_tracking_rviz_plugin/social_activities_display.hpp
+  ##include/spencer_tracking_rviz_plugin/human_attributes_display.hpp
```

Comment:
why are these left out?

Reply:
Hi @makokal ! I am working on the migration of these nodes, but I preferred do a small PR to easily the revision process.